back-end/src/models/AppUser/AppUser.repository.ts

```diff
@@ -163,6 +184,61 @@ export default class AppUserRepository extends AppUserDb {
     });
   }
 
+  static async updateUserPasswordWithToken(
+    token: string,
+    password: string
+  ): Promise<AppUser> {
+    // Check if token is valid
+    const userToUpdate = await this.getUserByToken(token);
+    if (!userToUpdate) {
+      throw new Error("Aucun utilisateur ne correspond à ce token.");
+    }
```

Comment:
You can delete this because you already checked it in getUserByToken() method.

Reply:
ok

Reply:
Fix after VD codereview for the backend is ok